golang/pkg/dagent/update/update.go

```diff
@@ -0,0 +1,183 @@
+package update
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/dyrector-io/dyrectorio/golang/internal/helper/docker"
+	"github.com/dyrector-io/dyrectorio/golang/internal/helper/image"
+	"github.com/dyrector-io/dyrectorio/golang/internal/util"
+	"github.com/dyrector-io/dyrectorio/golang/pkg/dagent/utils"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/client"
+
+	containerbuilder "github.com/dyrector-io/dyrectorio/golang/pkg/builder/container"
+)
+
+var _selfUpdateDeadline *int64
+
+func getContainerName(ctx context.Context, base string) (string, error) {
+	name := base
+	count := 0
+
+	for {
+		container, err := docker.GetContainerByName(ctx, nil, name)
+		if err != nil {
+			return base, err
+		}
+		if len(container.Names) == 0 {
+			return name, nil
+		}
+
+		count++
+		name = base + fmt.Sprint(count)
+	}
+}
+
+func findImageAndPull(ctx context.Context, imageName string) (string, error) {
+	exists, err := containerbuilder.ImageExists(ctx, nil, imageName)
```

Comment:
Please move this func out of containerbuilder. Internal would be a comfy home for it. :blush: